internal/server/server.go

```diff
@@ -58,9 +64,18 @@ func (s *Neo4jMCPServer) Start() error {
 	if err := s.registerTools(); err != nil {
 		return fmt.Errorf("failed to register tools: %w", err)
 	}
-	slog.Info("Started Neo4j MCP Server. Now listening for input...")
-	// Note: ServeStdio handles its own signal management for graceful shutdown
-	return server.ServeStdio(s.MCPServer)
+
+	switch s.config.TransportMode {
+	case config.TransportModeHTTP:
+		// Note: ServeHTTP handles its own signal management for graceful shutdown
+		return s.StartHTTPServer()
+	case config.TransportModeStdio:
+		slog.Info("Starting stdio server")
+		// Note: ServeStdio handles its own signal management for graceful shutdown
+		return server.ServeStdio(s.MCPServer)
+	default:
+		return fmt.Errorf("unsupported transport mode: %s", s.config.TransportMode)
```

Comment:
I was thinking to coerce in the config the transport mode and raise a validation error, but I'm not too opinionated to keep it as well!

Reply:
> I guess this makes it a required config right? 

No, in the way in which I thought this, we have a default one (STDIO), I would say!

> Should we default to stdio if it’s missing (typo maybe in the config?) or just return an error.

Strongly leaning, on if it's set, either by environment variable or arguments, then we validate it and raise the error in case of wrong input, if not we default to STDIO.
To me in this way, we have the best of the two world, expected behaviour, best ergonomic!